<launch>
  <arg name="node_name" default="blob_detection" />

  <arg name="image" default="image" doc="The image topic. Should be remapped to the name of the real image topic." />

  <arg name="use_camera_info" default="false" doc="Indicates that the camera_info topic should be subscribed to to get the default input_frame_id. Otherwise the frame from the image message will be used." />
  <arg name="debug_view" default="true" doc="Specify whether the node displays a window to show debug image" />
  <arg name="queue_size" default="3" doc="Specigy queue_size of input image subscribers" />


  <arg name="hue_lower_limit" default="0" doc="The minimum allowed field value Hue" />
  <arg name="hue_upper_limit" default="179" doc="The maximum allowed field value Hue" />
  <arg name="sat_lower_limit" default="0" doc="The minimum allowed field value Saturation" />
  <arg name="sat_upper_limit" default="255" doc="The maximum allowed field value Saturation" />
  <arg name="val_lower_limit" default="0" doc="The minimum allowed field value Value" />
  <arg name="val_upper_limit" default="255" doc="The maximum allowed field value Value" />

  <arg name="morphology_ex_type" default="off" doc="off opening closing Specigy queue_size of input image subscribers" />
  <arg name="morphology_ex_kernel_size" default="3" doc="opening closing Specigy queue_size of input image subscribers" />
  <!-- blob_detection.cpp -->
  <node name="$(arg node_name)" pkg="opencv_apps" type="blob_detection" output="screen">
    <remap from="image" to="$(arg image)" />
    <param name="use_camera_info" value="$(arg use_camera_info)" />
    <param name="debug_view" value="$(arg debug_view)" />
    <param name="queue_size" value="$(arg queue_size)" />

    <param name="hue_lower_limit" value="$(arg hue_lower_limit)" />
    <param name="hue_upper_limit" value="$(arg hue_upper_limit)" />
    <param name="sat_lower_limit" value="$(arg sat_lower_limit)" />
    <param name="sat_upper_limit" value="$(arg sat_upper_limit)" />
    <param name="val_lower_limit" value="$(arg val_lower_limit)" />
    <param name="val_upper_limit" value="$(arg val_upper_limit)" />

    <param name="morphology_ex_type" value="$(arg morphology_ex_type)" />
    <param name="morphology_ex_kernel_size" value="$(arg morphology_ex_kernel_size)" />
  </node>
</launch>
